nixos/modules/services/torrent/deluge.nix

```diff
@@ -118,36 +118,78 @@ in {
             more informations.
           '';
         };
+
+        user = mkOption {
+          type = types.str;
+          default = "deluge";
+          description = ''
+            User account under which deluge runs.
+          '';
+        };
+
+        group = mkOption {
+          type = types.str;
+          default = "deluge";
+          description = ''
+            Group under which deluge runs.
+          '';
+        };
       };
 
       deluge.web = {
         enable = mkEnableOption "Deluge Web daemon";
+
+        user = mkOption {
```

Comment:
I can see how having a configurable user for the deluge daemon makes sense, but what's the purpose of this for the web daemon?

Reply:
Personally, I don't have a need for both daemons to run as different users, but I figured it was easy enough to do and it might be useful for someone. 

Once example that I can think of: I've got a bunch of iptables rules to mark packets from the deluge user, I might want to run the web interface as a different user to the `deluged` daemon so that the web interfaces' packets are not marked.

Reply:
I'd rather keep the option set limited to what people actually need. If a big need for these options arises at a later time we can still add them.

Reply:
And even so, the web user can default to the daemon user, so you can change both to the same value with the single option `services.deluge.user`, which should be enough.

Reply:
Having thought about this slightly more, I think keeping these as separate options would be valuable. 

In my previous example, marked packets from a given user might be sent over a VPN (which would be a common use case for Deluge, being a torrent client), and it would be desirable to have the packets from the web interface not be sent over the VPN (so that users can access this interface), and one easy way to solve this would be to have the web interface run as a non-VPN user (and thus not have its packets marked).

Reply:
Hm, does it actually work though? Because the web client seems to use the same config directory as the main daemon. If it's a different user, it might fail due to it not being able to read the directory.

Reply:
Ah, that's a good point, I hadn't considered that, I've removed the user/group options for the web daemon.

Reply:
On that note I'm gonna ask, did you make sure the `cfg.user/cfg.group` options work by setting them to a different user?

Reply:
Locally I’ve changed `cfg.group` and seen the change reflected in the unit file, Deluge still works and the downloaded files from it also have that group. I’ve not tested `cfg.user`.

Reply:
Should be fine yeah